src/toil/jobStores/aws/jobStore.py

```diff
@@ -1255,7 +1268,14 @@ def __getBucketRegion(self, bucket):
     def destroy(self):
         # FIXME: Destruction of encrypted stores only works after initialize() or .resume()
         # See https://github.com/BD2KGenomics/toil/issues/1041
-        self._bind(create=False, block=False)
+        try:
+            self._bind(create=False, block=False)
+        except BucketLocationConflictException:
+            # If the jobstore bucket existed, _bind would have raised BucketLocationConflictException
```

Comment:
Maybe add a comment in _bind() to alert the developer of this ordering sensitivity.